docs/developer/best-practices/typescript.asciidoc

```diff
@@ -30,6 +30,8 @@ This architecture imposes several limitations to which we must comply:
 Since project refs rely on generated `d.ts` files, the migration order does matter. You can migrate your plugin only when all the plugin dependencies already have migrated. It creates a situation where commonly used plugins (such as `data` or `kibana_react`) have to migrate first.
 Run `node scripts/find_plugins_without_ts_refs.js --id your_plugin_id` to get a list of plugins that should be switched to TS project refs to unblock your plugin migration.
 
+Additionally, in order to migrate into project refs, you also need to make sure your plugin doesn't have circular dependencies with other plugins. We run a job in the CI for each PR trying to find if new circular dependencies are being added which runs our tool with `node scripts/find_plugins_with_circular_deps`. However there are also a couple of circular dependencies already identified and that are in an allowed list to be solved. You also need to make sure your plugin don't rely in any other plugin into that allowed list. For a complete overview of the circular dependencies both found and in the allowed list as well as the complete circular dependencies path please run the following script locally with the debug flag `node scripts/find_plugins_with_circular_deps --debug` .
+
```

Comment:
Could you add a note that this restriction is applicable for both code & type imports? 
Probably it worth mentioning in the Kibana style guide as well?

Reply:
👍